samza-core/src/main/java/org/apache/samza/clustermanager/ContainerProcessManager.java

```diff
@@ -104,25 +105,26 @@
    */
   private final ClusterResourceManager clusterResourceManager;
 
-  /**
-   * If there are too many failed container failures (configured by job.container.retry.count) for a
-   * processor, the job exits.
-   */
-  private volatile boolean tooManyFailedContainers = false;
-
   /**
    * Exception thrown in callbacks, such as {@code containerAllocator}
    */
   private volatile Throwable exceptionOccurred = null;
 
+  private ContainerProcessManagerMetrics containerProcessManagerMetrics;
+  private JvmMetrics jvmMetrics;
+  private Map<String, MetricsReporter> metricsReporters;
+
   /**
    * A map that keeps track of how many times each processor failed. The key is the processor ID, and the
    * value is the {@link ProcessorFailure} object that has a count of failures.
    */
-  private final Map<String, ProcessorFailure> processorFailures = new HashMap<>();
-  private ContainerProcessManagerMetrics containerProcessManagerMetrics;
-  private JvmMetrics jvmMetrics;
-  private Map<String, MetricsReporter> metricsReporters;
+  final Map<String, ProcessorFailure> processorFailures = new HashMap<>();
+
+  /**
+   * If there are too many failed container failures (configured by job.container.retry.count) for a
+   * processor, the job exits.
+   */
+  volatile boolean tooManyFailedContainers = false;
```

Comment:
Can we expose a getter instead for testing?

Reply:
Will change.